Adds procedures for configuring private storage endpoints with Azure (#69024)
openshift/openshift-docs: modules/disabling-redirect-private-storage-endpoint-azure.adoc

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-private-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="disabling-redirect-private-storage-endpoint-azure"]
= Optional: Disabling redirect when using a private storage endpoint on Azure

> By default, redirect is enabled when using the image registry with a private storage endpoint on Azure. This ensures that users within the same cluster can pull images from the registry. 

Reviewer: Redirect is always enabled by default, not only when using a private storage endpoint on Azure. Up to you if we should keep the phrasing tho.

Reviewer: > This ensures that users within the same cluster can pull images from the registry.

Hmm this isn't quite true, redirect allows off-loading of traffic from the registry pods into the registry. This should make pulls faster.

Author: @flavianmissi sorry, below says spot on, and here it looks like you're giving suggestions :D How about this?

```suggestion
By default, redirect is enabled when using the image registry. Redirect allows off-loading of traffic from the registry pods into the registry, which makes pull faster. 

When redirect is enabled, users from outside of the cluster are unable to pull images from the registry. In some cases, users might want to disable redirect so that users from outside of the cluster can pull images from the registry. 

Use the following procedure to disable redirect.

```

Reviewer: looks good! the "spot on" comment was for the paragraph below "When redirect is enabled..." sorry for the confusion!

> When redirect is enabled, users from outside of the cluster are unable to pull images from the registry. In some cases, users might want to disable redirect so that users from outside of the cluster can pull images from the registry. 

Reviewer: Spot on!

Use the following procedure to disable redirect.

.Prerequisites

* You have configured an image registry to run on Azure

.Procedure

* Enter the following command to disable redirect on the image 
registry configuration:
+
[source,terminal]
----
$ oc patch configs.imageregistry cluster --type=merge -p '{"spec":{"disableRedirect": true}}'
----

.Verification

[NOTE]
====
The following procedure requires proper verification from a user outside of the cluster.
====

. Fetch the registry endpoint by running the following command:
+
[source,terminal]
----

> $ REGISTRY=$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}')

Reviewer: what if the user haven't configured a route? should a route be a prerequisite? or do we just note that somewhere?

Reviewer: no a route is not a prerequisite for making storage account private.

= Optional: Disabling redirect when using a private storage endpoint on Azure

By default, redirect is enabled when using the image registry. Redirect allows off-loading of traffic from the registry pods into the object storage, which makes pull faster. When redirect is enabled and the storage account is private, users from outside of the cluster are unable to pull images from the registry. 

In some cases, users might want to disable redirect so that users from outside of the cluster can pull images from the registry. 

Use the following procedure to disable redirect.

.Prerequisites

* You have configured the image registry to run on Azure
* You have configured a route.

.Procedure

* Enter the following command to disable redirect on the image 
registry configuration:
+
[source,terminal]
----
$ oc patch configs.imageregistry cluster --type=merge -p '{"spec":{"disableRedirect": true}}'
----

.Verification

[NOTE]
====
The following procedure requires proper verification from a user outside of the cluster.
====

. Fetch the registry service name by running the following command:
+
[source,terminal]
----
$ oc registry info
----
+
.Example output
+
[source,terminal]
----

> default-route-openshift-image-registry.<apps-crc.testing>

Author: Switched to <cluster_dns>

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-private-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="disabling-redirect-private-storage-endpoint-azure"]
= Optional: Disabling redirect when using a private storage endpoint on Azure

> By default, redirect is enabled when using the image registry. Redirect allows off-loading of traffic from the registry pods into the registry, which makes pull faster. When redirect is enabled, users from outside of the cluster are unable to pull images from the registry. 

Reviewer: ```suggestion
By default, redirect is enabled when using the image registry. Redirect allows off-loading of traffic from the registry pods into the object storage, which makes pull faster. When redirect is enabled and the storage account is private, users from outside of the cluster are unable to pull images from the registry. 
```

In some cases, users might want to disable redirect so that users from outside of the cluster can pull images from the registry. 

Use the following procedure to disable redirect.

.Prerequisites

* You have configured the image registry to run on Azure
* You have configured a route.

.Procedure

* Enter the following command to disable redirect on the image 
registry configuration:
+
[source,terminal]
----
$ oc patch configs.imageregistry cluster --type=merge -p '{"spec":{"disableRedirect": true}}'
----

.Verification

> +

Reviewer: nit: there is a `+` here, I think it's a typo?
```suggestion

```

[NOTE]
====

> The following procedure requires proper verification from a user outside of the cluster.

Reviewer: This sentence is correct, but the steps below are for verifying the procedure from within the cluster.
Remove the `oc debug node/<node-name>` and `chroot /host` steps and use the external registry route.

====

. Fetch the registry service name by running the following command:
+
[source,terminal]
----

> $ oc registry info --internal=true

Reviewer: When outside the cluster we need the external route, so no `--internal=true`.
```suggestion
$ oc registry info
```

----
+
.Example output
+
[source,terminal]
----

> image-registry.openshift-image-registry.svc:5000

Reviewer: This is not what the output will look like for the external route. On my OpenShift Local install this is what I see:
```
default-route-openshift-image-registry.apps-crc.testing
```
The `default-route-openshift-image-registry` part is consant, the rest will change depending on your cluster domain name.
